interfaces/molecule/ase.py

```diff
@@ -1,16 +1,84 @@
 from ...core.basemol import Molecule,Atom
+from ...core.functions import add_to_class
 from numpy import zeros as npz
 from numpy import array as npa
 
-__all__ = ['toASE', 'fromASE']
+__all__ = ['toASE', 'fromASE', 'readase', 'writease']
+ase_present = False
 
 try:
     from ase import Atom as aseAtom
     from ase import Atoms as aseAtoms
+    ase_present = True
 except ImportError:
     __all__ = []
 
 
+if ase_present:
+    @add_to_class(Molecule)
+    def readase(self, f, geometry, **other):
+        try:
+            from ase import io as aseIO
+        except ImportError:
+            raise MoleculeError('Asked for ASE IO engine but could not load ASE.io module')
+
+        aseMol = aseIO.read(f, **other)
+        mol = fromASE(aseMol)
+        #update self with the molecule read without overwriting e.g. settings
+        self += mol
+        #lattice does not survive soft update
+        self.lattice = mol.lattice
+        return
+
+
+    @add_to_class(Molecule)
+    def writease(self, f, **other):
+        aseMol = toASE(self)
+        aseMol.write(f, **other)
+        return
+
+
+    Molecule._readformat['ase'] = Molecule.readase
+    Molecule._writeformat['ase'] = Molecule.writease
+
+
+    #Hack to get documentation at the right place
```

Comment:
This is a really ugly hack :/
I think it would be much better to just "manually" put these two functions in the `molinterfaces.rst` file using regular sphinx `.. function::` directive instead of injecting it into `.py` file like that. In the end autodoc's `.. autofunction::` (and `.. automodule::`) directives are just wrappers that extract docstrings from source and produce a regular `.. function::` or `.. class::` entries.

Reply:
The question I got there: I thought that since the functions are only really existent if ASE is available, that they might not get included in the docs if the docs are built on a machine without ASE? That would be bad, since there would me multiple different docs then... Or am I seeing this wrong?

Reply:
I think we're overthinking modularity here ;) Someone building doc locally (srsly, who does that? :P) getting ASE documentation without ASE installed is much much less of a problem than SCM website not having ASE documentation, because something went wrong with ASE on the docbuilding bot.